### PR TITLE
Update golangci-lint version and fix linter errors

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
This PR updates golangci-lint to the version that other plugins use and fixate that version for CI.

It also fixes the newly found linter issues. A lot of them are formatting one due to the go1.18 doc updates.

#### Ticket Link
None

### CODE REVIEW

File: test/hello-world/hello.go [hunk 1]
```diff
@@ -131,9 +131,15 @@ func main() {
...
+	server := &http.Server{
+		Addr:              addr,
+		ReadHeaderTimeout: 3 * time.Second,
+		WriteTimeout:      5 * time.Second,
+	}
+
 	fmt.Println("Listening on", addr)
 	fmt.Println("Use '/apps install http http://localhost" + addr + "/manifest.json' to install the app") // matches manifest.json
-	log.Fatal(http.ListenAndServe(addr, nil))
+	panic(server.ListenAndServe())
```
Comment (hanzei) [inline test/hello-world/hello.go:142]: I went with `panic` as  that is was the other apps are also using. 0/5

**Review by javaguirre — APPROVED** [review]
LGTM! 👏

File: server/httpin/admin.go [hunk 2]
```diff
@@ -15,17 +15,22 @@ import (
...
-//      "add_deploys": []string e.g. ["aws_lambda","http"]
-//      "remove_deploys": []string e.g. ["aws_lambda","http"]
-//   Output: The updated listing manifest
+//
+//  1. The "core" manifest (except Deploy) is updated to the new values.
+//  2. Deploy types from the previously listed manifest are updated from the new manifest, or preserved.
+//  3. Deploy types specified in "add_deploys" are copied from the new manifest.
+//  4. "remove"
+//
+// _
```
Comment (levb) [inline server/httpin/admin.go:24]: What does this do?

**Review by levb — APPROVED** [review]

File: server/httpin/admin.go (hunk 2 above)
Comment (hanzei) [reply server/httpin/admin.go:24]: It separates the code block from the numbered list. Without the character the two blocks get combined, and the code block becomes part of the 4. item.

**Pull request merged**